libnavigation-core/src/test/java/com/mapbox/navigation/core/internal/utils/RouterExTest.kt

```diff
@@ -0,0 +1,72 @@
+package com.mapbox.navigation.core.internal.utils
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.fake.pckg.name.RouterDummyPackageName
+import com.mapbox.navigation.base.route.RouteRefreshCallback
+import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.RouterCallback
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class RouterExTest {
+
+    @Test
+    fun testIsInternalImplementation() {
+        listOf<Triple<String, Router, Boolean>>(
+            Triple(
+                "mock object creates with package name of original class/interface",
+                mockk(),
+                true
+            ),
+            Triple(
+                "anonymous kotlin class object creates with package name " +
+                    "where it's created",
+                createRouter(),
+                true
+            ),
+            Triple(
+                "external class creates in its own package",
+                createExternalRouter(),
+                false
+            ),
+        ).let { testCases ->
+            testCases.forEach { (message, router, isInternal) ->
+                assertEquals(message, isInternal, router.isInternalImplementation())
+            }
+        }
+    }
```

Comment:
What if split this test in 3?
```suggestion
    @Test
    fun `mock object creates with package name of original class interface`() {
        assertTrue(mockk<Router>().isInternalImplementation())
    }
    
    @Test
    fun `anonymous kotlin class object creates with package name where it's created`() {
        assertTrue( createRouter().isInternalImplementation())
    }
    
    @Test
    fun `external class creates in its own package`() {
        assertTrue(createExternalRouter().isInternalImplementation())
    }
```

Reply:
What do you mean by they are the same?

Reply:
I mean no difference, it's just different approaches